src/legacy/core_plugins/data/server/lib/get_search_params.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { first, map } from 'rxjs/operators';
+import { Request } from 'hapi';
+import KbnServer from 'src/legacy/server/kbn_server';
+import { SearchParams } from 'elasticsearch';
+import { SearchOptions } from '../../common';
+
+export async function getSearchParams(
+  request: Request,
+  searchParams: SearchParams,
+  options: SearchOptions
+) {
+  return {
+    ...searchParams,
+    timeout: searchParams.hasOwnProperty('timeout')
+      ? searchParams.timeout
+      : await getShardTimeout(request),
+    preference: searchParams.hasOwnProperty('preference')
+      ? searchParams.preference
+      : await getPreference(request, options),
+    maxConcurrentShardRequests: searchParams.hasOwnProperty('maxConcurrentShardRequests')
+      ? (searchParams as any).maxConcurrentShardRequests
```

Comment:
Any idea why `maxConcurrentShardRequests` isn't in the SearchParams type?  Is it an oversight in the typing or is this a kibana specific param?

Reply:
actually I see it's only part of msearch.  I guess for search endpoint it's irrelevant.